src/mprk.jl

```diff
@@ -24,19 +24,29 @@ function build_mprk_matrix(P, sigma, dt)
 end
 
 # in-place for dense arrays
-function build_mprk_matrix!(M, P, sigma, dt)
+function build_mprk_matrix!(M, P, sigma, dt; d = nothing)
     # M[i,i] = (sigma[i] + dt * sum_j P[j,i]) / sigma[i]
     # M[i,j] = -dt * P[i,j] / sigma[j]
     # TODO: the performance of this can likely be improved
     Base.require_one_based_indexing(M, P, sigma)
     @assert size(M, 1) == size(M, 2) == size(P, 1) == size(P, 2) == length(sigma)
+    if !isnothing(d)
+        Base.require_one_based_indexing(d)
+        @assert length(sigma) == length(d)
+    end
 
     zeroM = zero(eltype(P))
 
     # Set sigma on diagonal
     @inbounds for i in eachindex(sigma)
         M[i, i] = sigma[i]
     end
+    # Add nonconservative destruction terms to diagonal (PDSFunctions only!)
+    if !isnothing(d)
```

Comment:
This is the only difference when creating the matrix for a ConservativePDSFunction or a PDSFunction.